keanu-python/keanu/algorithm/optimization.py

```diff
@@ -42,14 +60,28 @@ def __init__(self,
                  relative_threshold: Optional[float] = None,
                  absolute_threshold: Optional[float] = None) -> None:
         builder, net = Optimizer._build_bayes_net(k.jvm_view().KeanuOptimizer.Gradient, net)
+        builder.algorithm(ConjugateGradient(max_evaluations, relative_threshold, absolute_threshold).unwrap())
```

Comment:
are you going to add `Adam` to python in this PR? 

if so, will you make `algorithm` configurable?

Reply:
I fixed what broke as a result in python but forgot to actually add the Adam class! I've added it to this file and made it configurable. I also added a few tests to make sure it can be used. Having just this file for optimization related code might not be a sustainable approach as we add more.